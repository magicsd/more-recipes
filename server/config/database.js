const path = require('path');
/**
 * The different database configurations for different environments.
 */
const config = {
  development: {
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_DATABASE,
    host: process.env.DB_HOST,
    dialect: process.env.DB_DIALECT,
    logging: false,
    operatorsAliases: false,
  },
  test: {
    dialect: 'sqlite',
    storage: 'database.sqlite',
    logging: false,
    operatorsAliases: false,
  },
  production: {
    connection_uri: process.env.DATABASE_URL,
    logging: false,
    operatorsAliases: false,
  }
};

module.exports = config;
