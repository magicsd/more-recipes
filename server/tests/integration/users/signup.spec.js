import supertest from 'supertest';
import { User } from '../../../database/models';
import app from '../../../index';

describe('User Sign Up Tests', () => {
  test('Should register a new user', async () => {
    await User.destroy({ where: {} });

    const fakeUser = {
      name: 'Alex Dus',
      password: 'password',
      email: 'sd@alexdus.com',
    };

    const { status, body } = await supertest(app)
      .post('/api/v1/users/signup')
      .send(fakeUser);

    expect(status).toBe(200);
    expect(body.data.user.name).toBe(fakeUser.name);
    expect(body.data.user.email).toBe(fakeUser.email);
    expect(body.data.access_token).toBeTruthy();

    const userFromDatabase = await User.find({ where: { email: fakeUser.email } });

    expect(userFromDatabase).toBeTruthy();
  });

  test('Should return validation error for duplicate email', async () => {
    const userData = {
      name: 'Alex Dus',
      password: 'password',
      email: 'sd@alexdus.com',
    };

    await User.destroy({ where: {} });

    await User.create(userData);

    const { body, status } = await supertest(app)
      .post('/api/v1/users/signup')
      .send(userData);

    expect(status).toBe(422);
    expect(body).toMatchSnapshot();
  });
});
