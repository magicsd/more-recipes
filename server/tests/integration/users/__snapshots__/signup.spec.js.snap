// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Sign Up Tests Should return validation error for duplicate email 1`] = `
Object {
  "data": Object {
    "errors": Array [
      "A user with this email already exists.",
    ],
  },
  "status": "fail",
}
`;
